### PR TITLE
Make it work with VSCode

### PR DESCRIPTION
**Pull request by kanocz**
few changes to make it work good with vscode

### CODE REVIEW

**Comment by acroca** [comment]
Can you add more information about the change? What's the problem? Is there any vscode issue related to this change?

**Comment by kanocz** [comment]
@acroca just try to install go extension in vscode and you'll see that it's no workspace symbols in `#` sign... I've traced and see that vscode sends only one argument - path to current project and `go-symbols` always returns empty list (something like `[]`)... but it's because we're ignoring `""` variant of relative folder which is project folder
+ I've added struct names before functions to know get something like `Type1.String` and `Type2.String` instead of many just `String` names without knowledge to which type it belong

File: main.go [hunk 1]
```diff
@@ -60,11 +61,27 @@ func (v *visitor) Visit(node ast.Node) bool {
...
 	var ident *ast.Ident
 	var kind string
+	var structName string
+
 	switch t := node.(type) {
 	case *ast.FuncDecl:
 		kind = "func"
 		ident = t.Name
 		descend = false
+		if nil != t.Recv {
```
Comment (johanbrandhorst) [inline main.go]: Could you invert this comparison? It's very unusual to compare in this way.

```
if t.Recv != nil {
```

File: main.go [hunk 2]
```diff
@@ -60,11 +61,27 @@ func (v *visitor) Visit(node ast.Node) bool {
...
+			if 1 == len(t.Recv.List) {
```
Comment (johanbrandhorst) [inline main.go]: This too

```
if len(t.Recv.List) == 1 {
```

File: main.go [hunk 3]
```diff
@@ -60,11 +61,27 @@ func (v *visitor) Visit(node ast.Node) bool {
...
+					structName = xt.Name + "."
+				case *ast.StarExpr:
+					switch xt2 := xt.X.(type) {
+					case *ast.Ident:
+						structName = xt2.Name + "."
+					}
+				}
+
+			}
+		}
```
Comment (johanbrandhorst) [inline main.go:86]: Could you add a comment explaining what this block does?

File: main.go [hunk 4]
```diff
@@ -156,7 +173,8 @@ func allPackages(ctxt *build.Context, sema chan bool, root string, ch chan<- ite
 		sema <- true
 		files, err := ioutil.ReadDir(dir)
 		<-sema
-		if pkg != "" || err != nil {
+		if err == nil {
+			// if pkg != "" || err != nil {
```
Comment (johanbrandhorst) [inline main.go]: Why was this changed? Is it valid to send an item with an empty `pkg` now?

File: main.go [hunk 5]
```diff
@@ -212,9 +230,9 @@ func doMain() error {
 	// Here we can't use buildutil.ForEachPackage here since it only considers
 	// src dirs and this tool should be able to run against a golang source dir.
 	forEachPackage(&ctxt, func(path string, err error) {
-		if path == "" {
-			return
-		}
+		// if path == "" {
+		// 	return
+		// }
```
Comment (johanbrandhorst) [inline main.go]: Why was this changed? Please remove it instead of just uncommenting it.

Comment (kanocz) [reply main.go]: it's not empty package, but "empty relative path" to path specified as argument... so this allows if `/home/user/go/src/github.com/acroca/go-symbols` specified as path for `go-symbols` list symbols in this folder and subfolders, not only in subfolders

**Comment by kanocz** [comment]
added commit with style corrections and comments :)

File: main.go (hunk 4 above)
Comment (johanbrandhorst) [reply main.go]: You didn't answer this question

File: main.go (hunk 5 above)
Comment (johanbrandhorst) [reply main.go]: Thanks, that makes sense.

**Comment by johanbrandhorst** [comment]
Thanks @kanocz, I assume you've tested this in some different cases? I know this is a tiny library and has no existing testing infrastructure so just knowing that you've tested it would be good.

File: main.go (hunk 4 above)
Comment (kanocz) [reply main.go]: It's the same as next one - if we'll not allow empty here next one never get top folder also

Comment (johanbrandhorst) [reply main.go]: Thanks

**Comment by kanocz** [comment]
@johanbrandhorst tested it with vscode both on Linux and osX with go 1.11 :)

**Review by johanbrandhorst — APPROVED** [review]
